app/models/notification.rb

```diff
@@ -55,56 +53,51 @@ def self.all_flavors
   #   naming::    Naming that triggered this email.
   #
   def calc_note(args)
-    if template = note_template
-      case flavor.to_sym
-      when :name
-        tracker  = user
-        observer = args[:user]
-        naming   = args[:naming]
-        unless observer
-          raise "Missing 'user' argument for #{flavor} notification."
-        end
-        unless naming
-          raise "Missing 'naming' argument for #{flavor} notification."
-        end
+    return unless (template = note_template)
 
-        template.
-          gsub(":observer", observer.login).
-          gsub(":observation", "#{MO.http_domain}/#{naming.observation_id}").
-          gsub(":mailing_address", tracker.mailing_address || "").
-          gsub(":location", naming.observation.place_name).
-          gsub(":name", naming.format_name)
+    case flavor.to_sym
+    when :name
+      tracker  = user
+      observer = args[:user]
+      naming   = args[:naming]
+      unless observer
+        raise "Missing 'user' argument for #{flavor} notification."
+      end
+      unless naming
+        raise "Missing 'naming' argument for #{flavor} notification."
       end
+
+      template.
+        gsub(":observer", observer.login).
+        gsub(":observation", "#{MO.http_domain}/#{naming.observation_id}").
+        gsub(":mailing_address", tracker.mailing_address || "").
+        gsub(":location", naming.observation.place_name).
+        gsub(":name", naming.format_name)
     end
   end
 
-  # Return principle target involved.  Again, this is different for each
+  # Return principal target involved.  Again, this is different for each
   # flavor:
   #
   # name::   Name that User is tracking.
   #
   def target
-    result = nil
-    if @target
-      result = @target
-    else
-      case flavor
-      when :name
-        result = Name.find(obj_id)
-      end
-      @target = result
-    end
-    result
+    # target already known
+    return @target if @target
+
+    @target = case flavor
```

Comment:
Again this seems like a legit use for `? :`.
`(flavor == :name ? Name.find(obj_id) : nil)`

Case statements with a single case strike me as odd and I don't like not expressing the default value.

Reply:
**Ternary operator**. Agree.

**Single-case `case`**. I also scratched my head about this, but decided to leave it alone:
1. I generally tried to minimize changes so that there was less to review;
2. The `case` statement makes it easier to add more notification flavors. 
3. There are 2 other single-case `case` expressions in this file. I think that if one changes form `case` to `if`, the others should also change. 
I'll change all three.
If, on reflection, you'd prefer the single-case `case`, just let me know.

**Unexpressed default**.  There's a special case when the default is `nil`. RuboCop complains  "Style/EmptyElse: Redundant `else`-clause" when it sees:
```ruby
if cond
  xyz
else
  nil
end
```
See https://rubocop.readthedocs.io/en/latest/cops_style/#styleemptyelse
So I'll have to do something like:
```ruby
result = nil
if cond
  result = xyz
end
result
``` 
Alternatively I could change the cop's EnforcedStyle to Empty (versus Both, which is the default).

To move things forward, I'll do the first (`result = nil`). If you change your mind, let me know and I'll go back to the unexpressed `nil`.

Reply:
Not a big deal.  I'd rather get rid of the single case `case` expressions.  I'm fine with an implicit default of `nil` if rubocop insists.

Reply:
I forgot to say that I was able to shorten this to be a one-line ternary if, with an expressed `nil`.
```ruby
  def target
    @target ||= flavor == :name ? Name.find(obj_id) : nil
  end
 ```

Reply:
My above comment is unclear.
I **did** get rid of the single case `case`, replacing 11 lines of code with this single line (including an expressed `nil`).
```ruby
@target ||= flavor == :name ? Name.find(obj_id) : nil
```
Is this too obscure?

Reply:
Works for me!